src/lib/workspace/AnalysisPanel.tsx

```diff
@@ -42,6 +42,17 @@ export function AnalysisPanel(props: Props) {
   const location = useLocation();
   const [lastVarPath, setLastVarPath] = useState('');
   const [lastVizPath, setLastVizPath] = useState('');
+  // use relativePath as input arg: 'Browse and Subset' do not have UUID
+  const isUUIDUrl = useCallback(
```

Comment:
Since `isUUIDUrl` is only being called once (to determine if a fullscreen visualization is being viewed), you can replace `useCallback` with `useMemo`, and rename `isUUIDUrl` to something like `inFullscreenVisualization`.

Maybe something like:

```typescript
const inFullscreenVisualization = useMemo(() => {
  const relativePath = location.pathname.replace(routeBase, '');
  const lastUrlElement = relativePath.split('/').pop();

  // 'Browse and Subset' and the 'Visualize' selector do not have a UUID in the url
  return (
    lastUrlElement?.match(
      '^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'
    ) != null
  );
}, [location, routeBase]);
  ```

Reply:
Thank you for your quick review and comment @jtlong3rd 👍 I agree with your suggestion and changed to use useMemo instead of useCallback